### ISSUE TITLE
Ctrl+W is not consistent with Epiphany & Terminal

### ISSUE THREAD

**Issue by jlnr** [elementary/files#327]
When only one tab is open in Files, Ctrl+W does not close the window, but instead opens the home folder again. I think that Epiphany's/Terminal's behavior is more intuitive here - Ctrl+W with only one open tab closes the window.

**Comment by jeremypw** [comment]
If the design team is in agreement this is easily changed.

**Comment by danrabbit** [comment]
We previously decided against this because `Ctrl + W` is the close tab shortcut, not the close window shortcut. If your intent is the close the window there is another shortcut for that. But changing this would mean that we're trying to be clever and assume your intent instead of following what you told the computer to do.

For example, if your intent is simple to close all of your open tabs so that you have a clear "workspace" and you mash `Ctrl + W` and then suddenly Files quits, this looks like a crash and does something different from what you told Files to do (Close tabs).

**Comment by cassidyjames** [comment]
So, this is actually a bug with Terminal and Epiphany. 😉

**Comment by jlnr** [comment]
There's some predictable consistency in the Epiphany-like behavior that I enjoy: If you press Ctrl+W when only one tab is open, the window closes. Likewise, if you press Alt+F4 when only one window is open, the application quits.

Other apps like Scratch/Code and most "workbench apps" (where mashing Ctrl+W is especially common) don't close the window when the last tab is closed (or dragged out from the tab bar). Instead, they display an empty state screen with an empty tab bar, which ignores further Ctrl+W presses and typically makes it easy to get started with the next project.

Files is an outlier because it neither closes the window, nor does it display an empty state screen. It immediately re-opens a new tab with the home folder, and in a way, this is also a guess at what I want to do next; it could also keep the user's last tab open, or show a list of recent locations below an empty tab bar.

I don't think any app's behavior is unreasonable, but I feel that changing Files to match Epiphany/Terminal would make the OS both more internally consistent and more compatible with muscle memory from third-party apps (all browsers seem to work like Epiphany now).

I guess this is the best case I can make for this :)

**Comment by jeremypw** [comment]
Elementary apps are supposed to restore their state at close when reopened.  If the last tab has been closed what is intended to be restored?  In fact Terminal restores the tab that was closed which seems illogical.  It seems more logical for closing the last tab to put the app in a sensible "initial" state.  In the case of Code then there is a sensible "no document" state but that is not the case for Files or Terminal. Showing the user home directory is not an unreasonable default state, however.

**Comment by artemanufrij** [comment]
@jeremypw 
first thing what I do after fresh install is to disable tab restoring for all apps.
currently it's not possible to disable it for code. It was the reason for me for switch to another editor.

**Comment by jlnr** [comment]
@jeremypw Terminal's restoration behavior is both practical and logical in my opinion: If you close the last tab of the last window, the tab will not be restored later, because the you meant to close the tab -- closing the window is only a side effect of that. However, if you exit the app by other means than closing all tabs and then relaunch it, the previous tab(s) will be restored.

I'm a big fan of tab/state restoration, but I still prefer the "Terminal model" and I think the two are perfectly compatible (I can't compare with Epiphany right now without losing my precious tab collection :) ).

**Comment by jeremypw** [comment]
@jlnr:  For me, closing the last tab in Terminal *does* restore that tab.  When you say "the tab will not be restored later", what does happen and what do you expect?  Have you turned off tab restoration in settings?

**Comment by jeremypw** [comment]
It does seem reasonable that at least Terminal and Files have the same behaviour as they are both show one folder per tab.  But I am not sure what that behaviour should be.

**Comment by jeremypw** [comment]
It would be interesting to know what proportion of users turn off tab restore (or would like to if they knew how).

**Comment by jeremypw** [comment]
@artemanufrij : Have you opened an issue in Code?  It seems inconsistent have this setting in some apps and not others.

**Comment by artemanufrij** [comment]
@jeremypw we could start a survey on G+

**Comment by jlnr** [comment]
What happens for me with Terminal (and I don't remember changing any settings):

1. Launch Terminal
2. `cd Pictures`
3. Press Shift+Ctrl+W, or middle-click the tab, or click the (x) in the tab
4. Launch Terminal again
5. I now have one tab in the default location (home folder)

1. Launch Terminal
2. `cd Pictures`
3. Press click the (x) in the window header, or quit Terminal via the context menu in Plank
4. Launch Terminal again
5. I now have one tab in ~/Pictures (tab restored)

**Comment by artemanufrij** [comment]
@jeremypw yes. https://github.com/elementary/code/issues/162

**Comment by jeremypw** [comment]
@jlnr That is not what I experience (using Terminal master compiled and installed on Juno) - are you on Juno or Loki?

**Comment by jlnr** [comment]
Ah sorry, I am still on Loki and hadn't tested these steps in my Juno VM. That probably explains the difference. I prefer the Loki behavior. Epiphany still behaves the way I described it (tabs are forgotten when explicitly closed).

**Comment by jeremypw** [comment]
@artemanufrij :  I did not realise this was a closed, "Won't Fix" issue in Code.  This being so, a survey may be interesting but would not necessarily influence elementary design.

**Comment by mahajanudit** [comment]
I agree with jlnr and others that Ctrl + W with last tab open should be intuitive. Currently the way it refreshes to home in an instant looks a bit buggy. Also, many apps, not just Gnome apps, but even outside of gnome or even linux for example web browsers, firefox and chrome have this behavior. Ctrl+w is much more habitual then Alt+F4 and most users are used to it.
We also need to think of what users generally do, most users spend their majority time in web browsing (given we want avg user to come to linux) so the behaviour should be consistent. These are my 2 cents

**Comment by eljefuri** [comment]
I am in agreement (as a user) that this current behavior of reloading the home window is quite strange. I expect it to close Files if it is on the last tab. For what it's worth I don't think that every application that features tabs should necessarily have identical behavior, but this at least mirrors my expectations coming from macOS and of how browsers work.

**Comment by sprite-1** [comment]
Sorry for butting into this discussion and this is slightly off-topic but here's a compromise I can think of for people who want their tabs restored and those that want a clean slate whenever they open an app. I'd just like to preface this by saying I'm one of the people whose first move after finding out about the tab restore feature is to find out how to disable it on both Terminal and Files.

Anyway, my suggestion is to make the default behavior be to _start fresh_ **but** use this button to easily restore previous session. That way it caters for both use cases.

![image](https://user-images.githubusercontent.com/33685433/50736122-26134400-11f4-11e9-8476-6ef2ddec4bb8.png)

**Comment by jeremypw** [comment]
@sprite-1  An interesting idea but I doubt the default behaviour will be changed as it is a key part of elementary design principles that apps open fast and restore their state.  There an open PR #823 that stops tabs being restored if History is off in the Privacy settings but some questions are outstanding on this behaviour and may be too blunt an instrument for some even if easier than fiddling with dconf settings.

**Comment by sprite-1** [comment]
@jeremypw I'll understand if the default behavior will not be changed to this but at the least, give us a flag we can toggle in dconf editor for it 👼

**Comment by jeremypw** [comment]
You can change the restore tabs behaviour by changing io/elementary/files/preferences/restore-tabs setting with dconf-editor or the commandline.

**Comment by sprite-1** [comment]
I've set that option long ago 👍 but what I meant is the option to restore the previous session without it being forced on me everytime I open the app

**Comment by jeremypw** [comment]
OK.  We are rather constrained by the embargo on new widgets in the Files interface but your suggestion may work.

**Comment by daquinons** [comment]
Please, I would suggest to take this in consideration. Ctrl W closes tabs, and when no tab is there, the application should close. This is the expected behaviour in all the software, including native apps in eOS. The behaviour is extremely unpleasant from the UX point of view.

**Comment by jeremypw** [comment]
@danirabbit As you have marked this as a "Bug" are you happy for me to go ahead and make Files match Terminal behaviour in this respect?

**Comment by danirabbit** [comment]
@jeremypw sorry I haven't read the discussion here I was mostly just sorting things out and hadn't started confirming them yet
